crates/common/certificate/src/lib.rs

```diff
@@ -146,6 +147,40 @@ impl KeyCertPair {
     }
 }
 
+pub fn translate_error(inner: &str) -> CertificateError {
```

Comment:
This function must take a `rustls::Error::InvalidCertificateData` parameter and not a `&str`. Doing so will not force the caller to extract the error message from the TLS error.

I would also rename this function as `translate_tls_error` or better implement `From<rustls::Error::InvalidCertificateData> for `CertificateError`.